core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -712,7 +712,7 @@ void populate1() {
       defineMethod(CONCAT_FUNCTION, BuiltInMethod.MULTI_STRING_CONCAT.method,
           NullPolicy.STRICT);
       defineMethod(CONCAT_FUNCTION_WITH_NULL,
-          BuiltInMethod.MULTI_STRING_CONCAT_WITH_NULL.method, NullPolicy.NONE);
+          BuiltInMethod.MULTI_STRING_CONCAT_WITH_NULL.method, NullPolicy.STRICT);
```

Comment:
this one seems wrong; see the JavaDoc for the corresponding function.

Reply:
The method might be tolerant of nulls inside the array... but the guard isn't for the elements within the array, it's for the array reference itself. If you pass a null array to {{concatMultiWithNull}} it will throw NPE.

Reply:
This is the JavaDoc:
```
  /** The "CONCAT(arg, ...)" function that concatenates strings,
   * which never returns null.
   * For example, "CONCAT('a', 'bc', 'd')" returns "abcd".
   *
   * <p>If one of the arguments is null, it will be treated as empty string.
   * "CONCAT('a', null)" returns "a".
   * "CONCAT('a', null, 'b')" returns "ab".
   *
   * <p>Returns empty string only when all arguments are null
   * or the empty string.
   * "CONCAT(null)" returns "".
   * "CONCAT(null, '')" returns "".
   * "CONCAT(null, null, null)" returns "".
   *
   * <p>It differs from {@link #CONCAT_FUNCTION} when processing
   * null values. */
  @LibraryOperator(libraries = {MSSQL, POSTGRESQL}, exceptLibraries = {REDSHIFT})
  public static final SqlFunction CONCAT_FUNCTION_WITH_NULL =
```

Reply:
So this function does not even have arrays as arguments, only strings, if the javadoc is correct.

Reply:
It may be that it's impossible to write a SQL command that will call this method with an actual `null` array reference... in which case maybe `NONE` is a safe approach here. If it is possible though then the contract would suggest that the guards should result in an empty string return, and not a null short circuit. If thats the case then there may be some configuration missing to allow to specify what the return value should be if a guard is triggered.

Reply:
if you can create an example that fails I will consider updating this, but otherwise I leave it as is

Reply:
Sorry, thats a :+1: to "I think this is probably impossible so lets leave it as is".